### PR TITLE
Upgrade to kotlin 1.8

### PR DESCRIPTION
**Pull request by sstone**

### CODE REVIEW

**Comment by dpad85** [comment]
I've been able to test this branch (locally merged with `splices-cpfp`) on Phoenix Android. A few libs had to be upgraded (e.g. SQLDelight to 1.3.5) to support kotlin 1.8, but otherwise there have been no issues.

We'll have to test on iOS as well but this looks good.

**Comment by dpad85** [comment]
Tests also look good on iOS. There are a few Swift-related changes that will need Robbie's touch, but no blockers. I think we can go ahead with kotlin 1.8 on lightning-kmp 👍

**Comment by sstone** [comment]
Nice! I'll publish released version of secp256k1-kmp and bitcoin-kmp and update this PR when it's done

File: build.gradle.kts [hunk 1]
```diff
@@ -14,7 +14,7 @@ allprojects {
     version = "1.5.0-SNAPSHOT"
 
     repositories {
-        mavenLocal()
+        //mavenLocal()
```
Comment (pm47) [inline build.gradle.kts:17]: Why is this commented?

File: publishing/lightning-kmp-snapshot-deploy.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash -x
```
Comment (pm47) [inline publishing/lightning-kmp-snapshot-deploy.sh:1]: What is the purpose of this file?

File: build.gradle.kts (hunk 1 above)
Comment (sstone) [reply build.gradle.kts]: I added a comment in https://github.com/ACINQ/lightning-kmp/pull/457/commits/4798b41df3534a750c0474746de9da69bdfe4488 (we could also just remove this line)

File: publishing/lightning-kmp-snapshot-deploy.sh (hunk 2 above)
Comment (sstone) [reply publishing/lightning-kmp-snapshot-deploy.sh:1]: To publish snapshot libraries to sonatype so they can be available globally (for other devs/machines, github CI, ...)

**Review by pm47 — APPROVED** [review]

**Comment by pm47** [comment]
Thanks!

**Pull request merged**